docs: add cluster-wide inhibition example to alerting tutorial (#3015)
prometheus/docs: docs/tutorials/alerting_based_on_metrics.md

<iframe width="560" height="315" src="https://www.youtube.com/embed/xaMXVrle98M" frameborder="0" allowfullscreen></iframe>

Similarly Alertmanager can be configured with other receivers to notify when an alert is firing.

## Inhibiting alerts from an entire cluster

> When a whole cluster (or instance) becomes unreachable, you usually don't want a separate notification for every alert that fires as a consequence. Alertmanager's [inhibition](/docs/alerting/latest/alertmanager/#inhibition) feature lets a single "cluster is down" alert mute all the dependent alerts coming from that same cluster, so you receive one meaningful notification instead of a flood.

Reviewer: ```suggestion
When an entire cluster becomes unreachable, you usually don't want separate notifications for every alert that fires as a consequence. Alertmanager's [inhibition](/docs/alerting/latest/alertmanager/#inhibition) feature lets a single, firing "cluster is down" alert mute all alerts coming from that same cluster, so you receive one meaningful notification instead of a flood.
```

Inhibition is configured with `inhibit_rules` in `alertmanager.yml`. The following rule mutes every alert that shares the same `cluster` label value as a firing `ClusterUnreachable` alert:

> alertmanager.yml

```yaml
inhibit_rules:
  - source_matchers:
      - 'alertname = "ClusterUnreachable"'
    target_matchers:
      - 'alertname != "ClusterUnreachable"'
    equal:
      - 'cluster'

> ```

Reviewer: Let's show users the full `alertmanager.yml` here, similarly to how this is done previously in this file. You can re-use the previous listing and just add the inihibit rule.

- `source_matchers` selects the alert that suppresses others when it is firing (here, `ClusterUnreachable`).

> - `target_matchers` selects the alerts to mute. `ClusterUnreachable` is excluded so the source alert itself is still delivered.

Reviewer: Below it says the triggering alert is always delivered. Maybe its worth quoting the docs here: 

> To prevent an alert from inhibiting itself, an alert that matches both the target and the source side of a rule cannot be inhibited by alerts for which the same is true (including itself). However, we recommend to choose target and source matchers in a way that alerts never match both sides. It is much easier to reason about and does not trigger this special case.
